libbeat/common/kubernetes/k8skeystore/kubernetes_keystore.go

```diff
@@ -101,6 +101,9 @@ func NewKubernetesSecretsKeystore(keystoreNamespace string, ks8client k8s.Interf
 func (k *KubernetesSecretsKeystore) Retrieve(key string) (*keystore.SecureString, error) {
 	// key = "kubernetes.somenamespace.somesecret.value"
 	tokens := strings.Split(key, ".")
+	if tokens[0] != "kubernetes" {
+		return nil, keystore.ErrKeyDoesntExists
+	}
```

Comment:
Unrelated change (and it should check if `tokens` has any elements before trying to get the first one :slightly_smiling_face:)